src/components/list/list.tsx

```diff
@@ -338,56 +339,56 @@ export class List implements InteractiveComponent, LoadableComponent {
     }
   };
 
+  private filterElements({
+    el,
+    filteredItems,
+    visibleParents,
+    visible
+  }: {
+    el: HTMLCalciteListItemElement | HTMLCalciteListItemGroupElement;
+    filteredItems: HTMLCalciteListItemElement[];
+    visibleParents: WeakSet<HTMLElement>;
+    visible?: boolean;
+  }): void {
+    const isVisible = filteredItems.includes(el as HTMLCalciteListItemElement) || visible;
```

Comment:
IMO, the `visible` override might make this slightly tricky to debug.

Reply:
Refactored!